### PR TITLE
Auth upgrade

### PR DESCRIPTION
**Pull request by hellgaynius**

### CODE REVIEW

File: src/settings/axiosSettings.js [hunk 1]
```diff
@@ -1,41 +1,101 @@
...
+  CREDENTIALS_ERROR: SIGNIN_ERROR_MESSAGE,
+  NOT_FOUND_USER_ERROR: SIGNIN_ERROR_MESSAGE,
+  DELETED_USER_ERROR: SIGNIN_ERROR_MESSAGE,
+
+  SIGNUP_WRONG_CODE_ERROR: WRONG_CODE_ERROR_MESSAGE,
+  SIGNUP_EXPIRED_CODE_ERROR: EXPIRED_CODE_ERROR_MESSAGE,
+
+  RESET_PASSWORD_WRONG_CODE_ERROR: WRONG_CODE_ERROR_MESSAGE,
+  RESET_PASSWORD_EXPIRED_CODE_ERROR: EXPIRED_CODE_ERROR_MESSAGE,
+};
```
Comment (vyadl) [inline src/settings/axiosSettings.js:26]: I think all of that should be in separated file and importing here

File: src/settings/axiosSettings.js [hunk 2]
```diff
@@ -1,41 +1,101 @@
...
-};
+}
+
+export function getErrorMessage(errorData) {
+  const { code, message } = errorData;
+
+  return ERROR_CODES_AND_MESSAGES_ACCORDANCE[code]
+    || message
+    || 'Something went wrong';
+}
```
Comment (vyadl) [inline src/settings/axiosSettings.js:93]: this also should be in separate module with all errors

File: src/store/auth/actions.js [hunk 3]
```diff
@@ -1,52 +1,207 @@
...
+        if (mode !== 'allExceptCurrent') {
+          commit('logOut');
+          localStorage.removeItem('user');
+          localStorage.removeItem('currentListId');
+          deleteAccessToken();
+          commitFromRoot('sidebar/closeSidebar');
+          commitFromRoot('filters/resetFilters');
+          commitFromRoot('setCurrentListItems', []);
+          dispatchFromRoot('visualization/_resetVisualizationToDefault');
+          router.push({ name: 'auth' });
```
Comment (vyadl) [inline src/store/auth/actions.js:194]: I would like to propose you to add some empty lines between logical groups of actions here

File: src/router/routes.js [hunk 4]
```diff
@@ -9,14 +15,80 @@ export const routes = [
...
+    path: '/auth/request-reset-password',
+    name: 'requestResetPassword',
+    components: {
+      logo: AppLogo,
+      form: RequestResetPassword,
+    },
     meta: {
       isLoginRequired: false,
+      isAuthPage: true,
     },
```
[the remark this answers is not in the document]
Comment (hellgaynius) [reply src/router/routes.js:92]: made trello card

File: src/utils/misc.js [hunk 5]
```diff
@@ -34,3 +34,13 @@ export function getFormattedDate(val) {
 
   return new Intl.DateTimeFormat('en', options).format(new Date(val));
 }
+
+export function handleErrorAndRequestStatus(request, values) {
```
Comment (vyadl) [inline src/utils/misc.js:38]: values is too abstract here

Comment (vyadl) [reply src/utils/misc.js:38]: maybe componentRequestFields?

**Pull request merged**